### PR TITLE
Re-enable immutable forwarder in ERC2771Context

### LINKED ISSUE #2905: Add back immutable in ERC2771Context
We had to remove use of immutable since the variable could not be read during the constructor, causing all sorts of issues as pointed out in https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2739.

Since Solidity 0.8.8 this should no longer be a problem (though we should confirm with a test).

### PR DESCRIPTION
**Pull request by Amxx**
Fixes #2905 

#### PR Checklist

- [x] Tests
- [ ] Documentation
- [x] Changelog entry

### CODE REVIEW

File: contracts/metatx/ERC2771Context.sol [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+pragma solidity ^0.8.9;
 
 import "../utils/Context.sol";
 
 /**
  * @dev Context variant with ERC2771 support.
  */
 abstract contract ERC2771Context is Context {
-    address private _trustedForwarder;
+    address private immutable _trustedForwarder;
```
Comment (frangio) [inline contracts/metatx/ERC2771Context.sol:12]: Do we want this to be immutable in Upgradeable Contracts?

Comment (Amxx) [reply contracts/metatx/ERC2771Context.sol:12]: I think so.

Comment (frangio) [inline contracts/metatx/ERC2771Context.sol:12]:
```suggestion
    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
    address private immutable _trustedForwarder;
```

**Comment by frangio** [comment]
Note: Missing changelog entry.

**Review by frangio — APPROVED** [review]

**Comment by frangio** [comment]
I want to wait to merge once we've finished the 4.4 release, because I realized there will be some issues around the immutable variable and the transpiler.

**Comment by frangio** [comment]
Should be good now for the transpiler I think.

**Pull request merged**